### PR TITLE
feat(gce): adds address reader service

### PR DESCRIPTION
**Pull request by danielpeach**
@jtk54 or @duftler please review.

### CODE REVIEW

File: app/scripts/modules/google/address/address.reader.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+class GceAddressReader {
+  static get $inject() { return ['searchService', 'infrastructureCaches']; }
+
+  constructor (private searchService: SearchService, private infrastructureCaches: InfrastructureCacheService) {}
+
+  public listAddresses(region?: string): IPromise<IGceAddress[]> {
+    if (region) {
+      return this.listAddresses().then(addresses => addresses.filter(address => address.region === region));
```
Comment (jtk54) [inline app/scripts/modules/google/address/address.reader.ts]: Can you add an explicit null and a comment in the recursive `listAddresses` to make it a bit clearer? Something like `listAddresses(null /* region */)` to make this a little less subtle.

File: app/scripts/modules/google/address/address.reader.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    } else {
+      return this.searchService
+        .search<IAddressSearchResults>({q: '', type: 'addresses'}, this.infrastructureCaches.get('addresses'))
+        .then((searchResults: ISearchResults<IAddressSearchResults>) => {
+          if (searchResults && searchResults.results) {
+            return searchResults.results.map(result => {
+              let address = JSON.parse(result.address) as IGceAddress;
+              address.account = result.account;
+              // An address's selfLink ends with its region.
```
Comment (jtk54) [inline app/scripts/modules/google/address/address.reader.ts]: Is this the case for global addresses?

File: app/scripts/modules/google/address/address.reader.ts (hunk 1 above)
Comment (danielpeach) [inline app/scripts/modules/google/address/address.reader.ts]: Sounds good.

File: app/scripts/modules/google/address/address.reader.ts (hunk 2 above)
Comment (danielpeach) [inline app/scripts/modules/google/address/address.reader.ts]: The `selfLink` ends with `'global'` for global addresses.

**Pull request merged**